pkg/p2consul/kv.go

```diff
@@ -0,0 +1,138 @@
+// Package p2consul provides a generalized API for reading and writing manifests
+// in consul.
+package p2consul
+
+import (
+	"bytes"
+	"strings"
+	"time"
+
+	"github.com/armon/consul-api"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/util"
+)
+
+type ManifestResult struct {
+	Manifest *pods.PodManifest
```

Comment:
This was a value, not a pointer, previously - as a best practice we are trying to avoid sending pointers over channels to mitigate potential race conditions if either side manipulates the value.